docs/docs-content/tutorials/getting-started/introduction-edge/build-edge-artifacts.md

```diff
@@ -0,0 +1,490 @@
+
+## Define Arguments
+
+Open a terminal window on your Linux machine and navigate to the **CanvOS** repository that you cloned in the
+[Prepare User Data for Edge Installation](./prepare-user-data.md) tutorial. This repository contains the utilities
+required to build the artifacts.
+
+```shell
+cd CanvOS
+```
+
+Check the available Git tags.
+
+```shell
+git tag
+```
+
+Check out the latest available tag. This tutorial uses the tag **v4.6.9** as an example.
+
+```
+git checkout v4.6.9
+```
+
+EdgeForge leverages [Earthly](https://earthly.dev/) to build the Installer ISO and provider images artifacts. The
+**.arg** file is used to pass the values of a few arguments, such as the image tag and registry name, to Earthly for the
+build process.
+
+Set a custom tag for the provider images. The tag must be an alphanumeric lowercase string. This tutorial uses
+`gs-tutorial` as an example.
+
+```bash
+export CUSTOM_TAG=gs-tutorial
+```
+
+Next, issue the following command to create the **.arg** file using the custom tag. The remaining arguments will use
+predefined values. For example, this tutorial uses [K3s](https://k3s.io/) as the Kubernetes distribution, Ubuntu as the
+OS distribution, and [ttl.sh](https://ttl.sh/) as the image registry.
+
+```bash
+cat << EOF > .arg
+CUSTOM_TAG=$CUSTOM_TAG
+IMAGE_REGISTRY=ttl.sh
+OS_DISTRIBUTION=ubuntu
+IMAGE_REPO=ubuntu
+OS_VERSION=22.04
+K8S_DISTRIBUTION=k3s
+ISO_NAME=palette-edge-installer
+ARCH=amd64
+UPDATE_KERNEL=false
+EOF
+```
+
+Verify that the file was created correctly using the `cat` command.
+
+```
+cat .arg
+```
+
+:::info
+
+Different versions of CanvOS may require different arguments. Refer to the
+[CanvOS](https://github.com/spectrocloud/CanvOS#readme) repository to learn more about the required arguments for each
+version tag.
+
+:::
+
+## Build Artifacts
+
+By default, Earthly builds multiple images with different Kubernetes versions. You can exclude the image versions you do
+not need from the build process by deleting the corresponding lines under the `k3s` section in the **k8s_version.json**
+file. This speeds up the build process and reduces the amount of space that is required from your host machine.
+
+Open the **k8s_version.json** file using an editor of your choice.
+
+```bash
+vi k8s_version.json
+```
+
+Next, delete the K3s versions you do not need. This tutorial uses K3s version `1.32.1` as an example. Below is an
+example of the file with all other versions removed.
+
+```text {18} hideClipboard
+{
+  "k3s": [
+    "1.32.1"
+  ],
+
+  ...
+
+}
+```
+
+Once you have made the alterations, save and exit the file.
+
+:::warning
+
+If you are using a CanvOS tag that is earlier than v4.4.12, the **k8s_version.json** file does not exist in those tags.
+In that case, open the **Earthfile** in the CanvOS directory. Under `build-provider-images` section, remove the lines
+containing the Kubernetes versions that you do not need.
+
+:::
```

Comment:
@caroldelwing It is worth noting that an environment variable named `K8S_VERSION` can be used to specify the Kubernetes version to build.

You can review step 6 in https://docs.spectrocloud.com/clusters/public-cloud/aws/eks-hybrid-nodes/prepare-environment/prepare-edge-hosts/#register-edge-host-in-appliance-mode for an example.

We also document it here: https://docs.spectrocloud.com/clusters/edge/edgeforge-workflow/palette-canvos/arg/#:~:text=%2C%20nodeadm.-,K8S_VERSION,-Kubernetes%20version.%20The